qwerty_ast/src/ast.rs

```diff
@@ -4,92 +4,104 @@
 
 // FIXME: Remove later!
 #![allow(unused_imports, dead_code)]
-use crate::basis::*;
-use crate::dimexpr::{DimExpr, DimVar, DimVarValue};
-use crate::inference::{ConstraintInfo, Context, DimConstraint};
+
 use crate::types::Type;
 use crate::types::*;
 use num::complex::Complex;
 use pyo3::exceptions::PyValueError;
 use pyo3::prelude::*;
-use pyo3::types::{PyFrame, PyString, PyList};
+use pyo3::types::{PyString, PyList};
 use std::collections::{HashMap, HashSet};
-pub use crate::basis::{QubitLiteral, QubitSymbol};
-use crate::dimexpr::DimExpr::Number;
 
 #[derive(Debug, Clone)]
 pub enum ASTNode {
-    QubitSymbol(QubitSymbol),
-    QubitLiteral(QubitLiteral),
+    ZeroQubit(ZeroQubit),
+    OneQubit(OneQubit),
```

Comment:
You don't even need these `ZeroQubit` and `OneQubit` structs in my opinion. We don't need for them to hold anything right now. Like you can just do:

```suggestion
    ZeroQubit,
    OneQubit,
```